replace derivative crate with derive_more crate

YJDoc2 opened the pull request:
This PR carries over from https://github.com/imsnif/bandwhich/pull/437 and changes the `derivative` crate to the `derive_more` crate. This is particularly for the `Debug` derives which were bandwhich's main usecase for `derivative` .

- Removed `debug_fn` (formatting fn) in code, as it only adds a fixed string, so directly used that string in the macro.
- Used Rust's normal `Default` derive, as our main use case for it was setting the default of `Verbosity` . But `Verbosity` itself is a simple struct having two `u8` fields and a `PhantomData` , which also does a `derive(Default)` as the default for `u8` is 0, the "normal" Default should be same as `Verbosity::new(0, 0)` .

cyqsimon commented:
Thanks for the PR; code looks pretty good. Two comments:

1. Seems like `derive_more` requires 1.75.0. I'm okay with it, but we do need to bump MSRV in `Cargo.toml`.
2. Please add a changelog entry.

Then we're good to merge.

cyqsimon commented on src/display/components/display_bandwidth.rs:
```diff
@@ -1,13 +1,13 @@
-
-use derivative::Derivative;
-
 use crate::cli::UnitFamily;
+use derive_more::Debug;
+use std::fmt;
```
Just a nit-pick: is this something your formatter did automatically? I think by default RA likes to order the imports in this order: `std`, foreign, `crate`.

Rustfmt doesn't enforce this (at least not yet), but I'd like to stick to RA's preferred formatting just in case. Also it's good for consistency.

YJDoc2 replied:
It might be my formatter, or it might be that I added the import in wrong way and ran cargo fmt manually, which didn't enforce this. In any case, I have updated the order.

YJDoc2 commented:
> Seems like derive_more requires 1.75.0. I'm okay with it, but we do need to bump MSRV in Cargo.toml.

Bumped `rust-version` in Cargo.toml

> Please add a changelog entry.

I have added two entries, one for crate change and once for msrv bump, both pointing to this PR, is that ok? Should I combine them in just one entry?

> Just a nit-pick: is this something your formatter did automatically? I think by default RA likes to order the imports in this order: std, foreign, crate.

Fixed, please check.

cyqsimon commented:
> I have added two entries, one for crate change and once for msrv bump, both pointing to this PR, is that ok? Should I combine them in just one entry?

Yeah either is fine really. Merging.
